File: src/test/java/com/homeclimatecontrol/esphome2influxdb/AutodiscoverTest.java
package com.homeclimatecontrol.esphome2influxdb;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.regex.Matcher;

import org.junit.jupiter.api.Test;

class AutodiscoverTest {

    @Test
    void sensorMatch() {

        var m = MqttReader.patternSensor.matcher("/esphome/7AC96F/sensor/1wire-bedroom-master-temperature/state");

        assertThat(m.matches()).isTrue();
        assertThat(m.group(1)).isEqualTo("/esphome/7AC96F");
        assertThat(m.group(2)).isEqualTo("1wire-bedroom-master-temperature");
    }
}

File: src/test/kotlin/com/homeclimatecontrol/esphome2influxdb/k/adapter/v1/in/AutodiscoverTest.kt
package com.homeclimatecontrol.esphome2influxdb.k.adapter.v1.`in`

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class AutodiscoverTest {
    @Test
    fun sensorMatch() {
        val m = MqttReader.patternSensor.matcher("/esphome/7AC96F/sensor/1wire-bedroom-master-temperature/state")
        Assertions.assertThat(m.matches()).isTrue
        Assertions.assertThat(m.group(1)).isEqualTo("/esphome/7AC96F")
        Assertions.assertThat(m.group(2)).isEqualTo("1wire-bedroom-master-temperature")
    }
}
